packages/react-components/src/trace-explorer/trace-explorer-time-range-data-widget.tsx

```diff
@@ -267,6 +268,48 @@ export class ReactTimeRangeDataWidget extends React.Component<
         }
     };
 
+    /**
+     * Allows public setting of active data and experiment data map.
+     * @param activeData
+     * @param map
+     */
+    public restoreState = (
+        activeData: ExperimentTimeRangeData,
+        experimentDataMap: Map<string, ExperimentTimeRangeData>
+    ): void => {
+        this.experimentDataMap = experimentDataMap;
+        this.setActiveExperiment(activeData);
+    };
+
+    private setActiveExperiment = (timeData?: ExperimentTimeRangeData): void => {
+        this.activeUUID = timeData?.UUID;
+        if (this.busy) {
+            return;
+        }
+        this.busy = true;
+        this.setState({ activeData: timeData }, () => {
+            this.busy = false;
+            if (this.activeUUID !== this.state.activeData?.UUID) {
+                const activeData = this.activeUUID ? this.experimentDataMap.get(this.activeUUID) : undefined;
+                this.setActiveExperiment(activeData);
+            }
+        });
+    };
+
+    get persistedState(): {
+        experimentDataMap: Map<string, ExperimentTimeRangeData>;
+        activeData?: ExperimentTimeRangeData;
+    } {
+        return {
+            experimentDataMap: this.experimentDataMap,
+            activeData: this.state.activeData
```

Comment:
Can't  you store the activeData as member and don't use the state here. Wouldn't it solve the timing issue?